libfwupdplugin/fu-device.c

```diff
@@ -1825,11 +1825,7 @@ fu_device_set_protocol (FuDevice *self, const gchar *protocol)
 void
 fu_device_set_physical_id (FuDevice *self, const gchar *physical_id)
 {
-	FuDevicePrivate *priv = GET_PRIVATE (self);
-	g_return_if_fail (FU_IS_DEVICE (self));
-	g_return_if_fail (physical_id != NULL);
-	g_free (priv->physical_id);
-	priv->physical_id = g_strdup (physical_id);
+	fwupd_device_set_physical_id (FWUPD_DEVICE (self), physical_id);
```

Comment:
I would recommend to instead change `fu-device.h` to map the function like other fu_* -> fwupd_* maps happen.

Reply:
I don't think we can do that, else we break plugin ABI as the symbol disappears. What Sjoerd did is right I think.

Reply:
Oh I see, good point.  I guess if/when plugin ABI breaks in the future that is's something we can remove then.